src/CommonLib/hostfxr_utility.cpp

```diff
@@ -369,3 +342,166 @@ HOSTFXR_UTILITY::SetHostFxrArguments(
     }
     return hr;
 }
+//
+// Invoke where.exe to find the location of dotnet.exe
+// Copies contents of dotnet.exe to a temp file
+// Respects path ordering.
+
+HRESULT
+HOSTFXR_UTILITY::FindDotnetExePath(
+    _In_ ASPNETCORE_CONFIG* pConfig,
+    _Out_ STRU* struDotnetPath)
+{
+    HRESULT             hr = S_OK;
+    STARTUPINFOW        startupInfo = { 0 };
+    PROCESS_INFORMATION processInformation = { 0 };
+    SECURITY_ATTRIBUTES securityAttributes;
+    STRU                struDotnetSubstring;
+    STRU                struDotnetLocationsString;
+    LPWSTR              pwzDotnetName = NULL;
+    DWORD               dwExitCode;
+    DWORD               dwNumBytesRead;
+    DWORD               dwFilePointer;
+    DWORD               dwBinaryType;
+    INT                 index = 0;
+    INT                 prevIndex = 0;
+    BOOL                fResult = FALSE;
+    BOOL                fIsWow64Process;
+    BOOL                fIsCurrentProcess64Bit;
+    BOOL                fFound = FALSE;
+    CHAR                pzFileContents[READ_BUFFER_SIZE];
+    HANDLE              hStdOutReadPipe = NULL;
+    HANDLE              hStdOutWritePipe = NULL;
+
+    securityAttributes.nLength = sizeof(securityAttributes);
+    securityAttributes.lpSecurityDescriptor = NULL;
+    securityAttributes.bInheritHandle = TRUE;
+
+    if (!CreatePipe(&hStdOutReadPipe, &hStdOutWritePipe, &securityAttributes, 0))
+    {
+        hr = HRESULT_FROM_WIN32(GetLastError());
+        goto Finished;
+    }
+    if (!SetHandleInformation(hStdOutReadPipe, HANDLE_FLAG_INHERIT, 0))
+    {
+        hr = ERROR_FILE_INVALID;
+        goto Finished;
+    }
+
+    // Set stdout and error to redirect to the temp file.
+    startupInfo.cb = sizeof(startupInfo);
+    startupInfo.dwFlags |= STARTF_USESTDHANDLES;
+    startupInfo.hStdOutput = hStdOutWritePipe;
+    startupInfo.hStdError = hStdOutWritePipe;
+
+    // CreateProcess requires a mutable string to be passed to commandline
+    // See https://blogs.msdn.microsoft.com/oldnewthing/20090601-00/?p=18083/
+    pwzDotnetName = SysAllocString(L"\"C:\\Windows\\System32\\where.exe\" dotnet.exe");
+
+    fResult = CreateProcessW(NULL,
+        pwzDotnetName,
+        NULL,
+        NULL,
+        TRUE,
+        CREATE_NO_WINDOW,
+        NULL,
+        NULL,
+        &startupInfo,
+        &processInformation
+    );
+
+    if (!fResult)
+    {
+        hr = ERROR_PROCESS_ABORTED;
+        goto Finished;
+    }
+
+    WaitForSingleObject(processInformation.hProcess, pConfig->QueryRequestTimeoutInMS());
+
+    fResult = GetExitCodeProcess(processInformation.hProcess, &dwExitCode);
```

Comment:
in case of child process hang or long running. closehandle will not kill the child process.

Reply:
what's the motivation for calling GetExitCodeProcess here if you don't want to check if the process is still active?

Reply:
Cause I forgot a check XD. Also I should be calling TerminateProcess too.